### PR TITLE
feat: revert-behavior

### PR DESCRIPTION
**Pull request by enitrat**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves: #574 

## What is the new behavior?

- implementing revert for EVM errors and REVERT opcode
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by greged93 — COMMENTED** [review]
Can we add a test that reverts in a subcontext and verify that the rest of the execution proceeds?
We can deploy two accounts, one with `call other_account, sstore 1` and `other_account` with `revert`, then check that the storage is correctly updated

**Review by greged93 — DISMISSED** [review]
lgtm

File: crates/evm/src/interpreter.cairo [hunk 1]
```diff
@@ -659,16 +659,24 @@ impl EVMInterpreterImpl of EVMInterpreterTrait {
     }
 
     /// Finalizes the revert of an execution context.
-    /// Clears all pending journal entries, not finalizing the pending state changes applied inside this context.
-    fn finalize_revert(ref self: Machine) { //TODO add the rest of the revert handling.
+    /// Clears all pending state journal entries, not finalizing the pending state changes applied inside this context.
+    fn handle_revert(ref self: Machine, maybe_message: Option<EVMError>) {
```
Comment (Eikix) [inline crates/evm/src/interpreter.cairo:663]: maybe_error?

Comment (enitrat) [reply crates/evm/src/interpreter.cairo:663]: maybe error message? idk

**Review by Eikix — APPROVED** [review]

**Pull request merged**